pkg/k8s/apps/statefulsets.go

```diff
@@ -104,6 +105,9 @@ func (i *StatefulSetApp) GetRevision() string {
 }
 
 func (i *StatefulSetApp) GetRunningPod(ctx context.Context, c kubernetes.Interface) (*apiv1.Pod, error) {
+	if i.sfs.Generation != i.sfs.Status.ObservedGeneration {
```

Comment:
When `ObservedGeneration` is updated, `status.UpdateRevision` is also updated, and we can check which pod is matching `status.UpdateRevision`